source/src/protocols/protein_grid/ProteinGrid.cc

```diff
@@ -0,0 +1,1493 @@
+ProteinGrid::ProteinGrid(core::pose::PoseOP in_pose, core::Real resolution)
+{
+	//seed initial values on member variables
+	initialize();
+
+	working_pose_ = in_pose;
+	resolution_ = resolution;
+	//validate the resolution
+	validate_resolution();
+
+	//reset the subregion vectors
+	//shift and bound vectors will be set/reset in the wrap function
+	reset_sub_region_vectors();
+
+	//wrap matrix around pose
+	wrap_matrix_around_pose();
+}
+
+//constructor that takes in pose and sub region vectors
+ProteinGrid::ProteinGrid(core::pose::PoseOP in_pose, numeric::xyzVector<int> sub_area_center, utility::vector1<core::Size> sub_region_dimensions)
+{
+	//seed initial values on member variables
+	initialize();
+
+	//set the working_pose_ to be the input pose
+	working_pose_ = in_pose;
+
+	//reset the subregion vectors
+	//shift and bound vectors will be set/reset in the wrap function
+	reset_sub_region_vectors();
+
+	//set dimensions
+	//throw a warning if the inputted subregion is not 3 entries long, will try to move forward anyway
+	if ( sub_region_dimensions.size() != 3 ) {
+		ms_tr.Warning << "Sub region dimensions vector that was provided does not have 3 expected values. Contents of vector: [";
+		for ( auto val : sub_region_dimensions ) {
+			ms_tr.Warning << val << ",";
+		}
+		ms_tr.Warning << "]" << std::endl;
+	}
+
+	//set values of true/absolute sub areas (center and dimensions before shifting)
+	true_sub_area_center_ = sub_area_center;
+	true_sub_region_dimensions_ = sub_region_dimensions;
+
+	using_sub_area_ = true;
+
+	//wrap matrix around pose
+	wrap_matrix_around_pose();
+}
+
+//constructor that takes in pose, resolution values, and subregion vectors
+ProteinGrid::ProteinGrid(core::pose::PoseOP in_pose, core::Real resolution, numeric::xyzVector<int> sub_area_center, utility::vector1<core::Size> sub_region_dimensions)
+{
+	//seed initial values on member variables
+	initialize();
+
+	working_pose_ = in_pose;
+	resolution_ = resolution;
+	//validate the resolution
+	validate_resolution();
+
+	//reset the subregion vectors
+	//shift and bound vectors will be set/reset in the wrap function
+	reset_sub_region_vectors();
+
+	//set dimensions
+	//throw a warning if the inputted subregion is not 3 entries long, will try to move forward anyway
+	if ( sub_region_dimensions.size() != 3 ) {
+		ms_tr.Warning << "Sub region dimensions vector that was provided does not have 3 expected values. Contents of vector: [";
+		for ( auto val : sub_region_dimensions ) {
+			ms_tr.Warning << val << ",";
+		}
+		ms_tr.Warning << "]" << std::endl;
+	}
+
+	//set values of true/absolute sub areas (center and dimensions before shifting)
+	true_sub_area_center_ = sub_area_center;
+	true_sub_region_dimensions_ = sub_region_dimensions;
+
+	using_sub_area_ = true;
+
+	//wrap matrix around pose
+	wrap_matrix_around_pose();
+}
+
+//copy constructor
+ProteinGrid::ProteinGrid( ProteinGrid const & other )
+{
+	initialize();
```

Comment:
If the clone function is just going to overwrite the initialized variables, then this call is pointless. And if the clone function is simply a straight member-by-member copy, it's much better to use the compiler-provided default one,  (`ProteinGrid::ProteinGrid( ProteinGrid const & other ) = default;` which has the benefit of being automatically updated whenever someone adds a new member variable.

Similar arguments apply for `operator=()` -- prefer the compiler default unless you have very special use case where it doesn't work.